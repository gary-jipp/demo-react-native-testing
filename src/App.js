import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import HomeStack from 'navigator/HomeStack';

const App = () => {
  return (
    <NavigationContainer>
      <HomeStack />
    </NavigationContainer>
  );
};

export default App;
