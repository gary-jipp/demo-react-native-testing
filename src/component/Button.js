import React from 'react';
import {StyleSheet, Text, View, Pressable, } from 'react-native';

const Button = (props) => {
  // console.log(`Button.${props.title}.disabled:`, props.disabled);

  return (
    <Pressable
      testID="button"
      accessibilityRole="button"
      disabled={props.disabled}
      style={props.disabled || styles.pressable}
      onPress={props.onPress} >
      <View style={[styles.button, props.style, props.disabled && styles.disabled]}>
        <Text style={[styles.text, props.textStyle]}>{props.title}</Text>
      </View>
    </Pressable >
  );
};

const styles = StyleSheet.create({
  pressable: function({pressed}) {
    return {opacity: pressed ? 0.8 : 1.0};
  },
  button: {
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: 12,
    paddingHorizontal: 20,
    borderRadius: 5,
    backgroundColor: '#2196F3',
    elevation: 0,
  },
  text: {
    fontSize: 16,
    lineHeight: 21,
    fontWeight: 'bold',
    letterSpacing: 0.25,
    color: '#EEE',
  },
  disabled: {
    backgroundColor: '#EEE',
  }
});

export default Button;