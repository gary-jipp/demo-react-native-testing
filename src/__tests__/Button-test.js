import 'react-native';
import React from 'react';
import Button from 'component/Button';
import App from 'App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('Button renders correctly with test-renderer', () => {
  const tree = renderer.create(<Button />);
  // console.log(tree.toJSON());
});

// Use built-in react test renderer
it('Button renders and matches snapshop', () => {
  const tree = renderer.create(<Button title="Press Me" />);
  const json = tree.toJSON();   // Docs recommend but doesn't seem to make a difference
  expect(tree).toMatchSnapshot(); // Creates snapshot on first run
});
