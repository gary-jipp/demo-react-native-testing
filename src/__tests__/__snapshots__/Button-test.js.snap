// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders and matches snapshop 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#2196F3",
          "borderRadius": 5,
          "elevation": 0,
          "justifyContent": "center",
          "paddingHorizontal": 20,
          "paddingVertical": 12,
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#EEE",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "lineHeight": 21,
          },
          undefined,
        ]
      }
    >
      Press Me
    </Text>
  </View>
</View>
`;
